File: tests/xcalablemp/not-implemented-yet/global-view/template/F/template_fix7.f90
      program main
      include 'xmp_lib.h'
!$xmp nodes p(4,4)
!$xmp template t(:,:,:)
!$xmp distribute t(*,cyclic(3),cyclic(7)) onto p
      integer N, s
      integer,allocatable:: a(:,:)
!$xmp align a(i,j) with t(*,i,j)
      character(len=2) result

      if(xmp_all_num_nodes().ne.16) then
         print *, 'You have to run this program by 16 nodes.'
      endif

      N = 0
      do i=1, 1000
         N = N+1
      enddo
      
!$xmp template_fix (*,cyclic(3), cyclic(7)) t(N,N,N)
      allocate(a(N,N))

!$xmp loop (i,j) on t(:,i,j)
      do j=1, N
         do i=1, N
            a(i,j) = xmp_node_num()
         enddo
      enddo
         
      s = 0
!$xmp loop (i,j) on t(:,i,j) reduction(+: s)
      do j=1, N
         do i=1, N
            s = s+a(i,j)
         enddo
      enddo

      result = 'OK'
      if(s .ne. 45000000) then
         result = 'NG'
      endif

      print *, xmp_all_node_num(), 'testp007.f ', result
      deallocate(a)

      end

File: tests/xcalablemp/NOT-IMPLEMENTED/303/template_fix7.c
#include<xmp.h>
#include<stdio.h>
#include<stdlib.h>     
#pragma xmp nodes p(4,4)
#pragma xmp template t(0:999,0:999,:)
#pragma xmp distribute t(*,cyclic(3),cyclic(7)) onto p
int i,j,N,s, **a,result=0;
#pragma xmp align a[i][j] with t(*,j,i)

int main(void)
{
  N = 1000;

#pragma xmp template_fix(*,cyclic(3),block(7)) t(N,N,N)
  for(i=0;i<N;i++)
    a[i]=(int *)malloc(sizeof(int) * N);

#pragma xmp loop (j,i) on t(:,j,i)
  for(i=0;i<N;i++)
    for(j=0;j<N;j++)
      a[i][j] = xmp_node_num();

   s = 0;
#pragma xmp loop (j,i) on t(j,i) reduction(+:s)
   for(i=0;i<N;i++)
     for(j=0;j<N;j++)
       s += a[i][j];
   
   result = 0;
   if(s != 79300)
     result = -1;

#pragma xmp reduction(+:result)
#pragma xmp task on p(1,1)
   {
     if(result == 0){
       printf("PASS\n");
     }
     else{
       fprintf(stderr, "ERROR\n");
       exit(1);
     }
   }

   for(i=0;i<N;i++)
     free(a[i]);
 
   return 0;
}
